spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/KerasLayer.scala

```diff
@@ -200,12 +224,24 @@ abstract class KerasLayer[A <: Activity: ClassTag, B <: Activity: ClassTag, T: C
    * @return node containing current module
    */
   override def inputs(nodes : Array[ModuleNode[T]]): ModuleNode[T] = {
-    Util.excludeNotKeras(nodes.map(_.element))
-    if (!nodes.isEmpty) { // as there's  Identity().inputs() within Graph
+    validateInput(nodes.map(_.element))
```

Comment:
What if we have:
```
l1 = n.inputs(n1)
l2 = n.inputs(n2)
```

Reply:
Humm, good catch. It's more about how to prevent stopping "shareWeights". Adding another flag to count? @yiheng any idea?

Reply:
Maybe we can have both usedAsSrc and usedAsDest or somthing similar.

Reply:
ok